typed-racket-lib/typed-racket/infer/infer-unit.rkt

```diff
@@ -720,8 +723,15 @@
           [((CustodianBox: t) (Evt: t*)) (cg S t*)]
           [((Channel: t) (Evt: t*)) (cg t t*)]
           [((Async-Channel: t) (Evt: t*)) (cg t t*)]
-          ;; we assume all HTs are mutable at the moment
-          [((Hashtable: s1 s2) (Hashtable: t1 t2))
+          [((Immutable-Hashtable: s1 s2)
+            (or (Hashtable: t1 t2)
+                (Immutable-Hashtable: t1 t2)))
+           ;; for immutable hash tables, covariant
+           (% cset-meet (cg s1 t1) (cg s2 t2))]
```

Comment:
The case where (IH A B) < (H A' B') doesn't seem right to me. I think that should be invariant.